### DISCUSSION TITLE
Infinite DName-Resolution

### DISCUSSION THREAD

**Discussion by AliveDevil** [TechnitiumSoftware/DnsServer#619] [Q&A] [answered]
So, here's my setup:
I have a zone dynamic.host.tld, where all DHCP leases are published to.
I have a @.host.tld DName record, pointing to dynamic.host.tld.
I have a @.domain.tld DName record pointing to host.tld.

This way I hoped, that I could resolve any DHCP hostname in dynamic.host.tld through both hostname.host.tld or hostname.domain.tld, without needing to create CName records for each and every dynamic host. This would also allow adding records to host.tld, which are automatically resolveable on domain.tld.

The issue is, that the Dns Server (11.1.1) seems to resolve for the shortest zone name available. A `dig` on hostname.domain.tld results in these records returned:
hostname.domain.tld CNAME hostname.host.tld
hostname.host.tld CNAME hostname.dynamic.host.tld
hostname.dynamic.host.tld CNAME hostname.dynamic.dynamic.host.tld
(continue infinitely)

For these records:
```
Zone domain.tld
@ DNAME host.tld

Zone host.tld
@ DNAME dynamic.host.tld

Zone dynamic.host.tld
hostname A 1.1.1.1
```

Is this wrong expectation on my side, or wrong implementation in the Dns server?

**Comment by ShreyasZare** [comment, marked as answer]
Thanks for that post. With DNAME, the entire sub-tree of the domain gets delegated to the configured domain name in the record. Which means you cannot have anything under that name in the `host.tld` zone. So, the config where you are trying to do a DNAME for a subdomain name wont work. The other DNAME record that you have in the `domain.tld` is fine though.

I tried the same config on my test setup and used the built-in DNS Client to test and it is resolving as expected. It seems that `dig` is capable of using DNAME record which is causing it to synthesize the infinite CNAME records on its own. The DNS Client built into the DNS server is not capable of processing DNAME so it ignores them and just process the CNAME records which is why it works with it.

I will add some kind of validation for this to prevent such config scenarios that can cause issues.

**Reply by AliveDevil** [reply]
Interesting. Thanks for the clarification.
